packages/utils/src/allOrCleanup.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Given an iterable of promises, settles them one at a time.
+ * If all of them resolve, returns the list of values.
+ * If one of them rejects, provides a callback for cleaning up the
+ * preceding (already resolved) values.
+ */
+export async function allOrCleanup<T>(
```

Comment:
Nice utility 👍 Could consider if we want to name this to e.g. to `promiseTransaction`? It either succeeds or fails, just like a DB transaction, and can be rolled back.

Reply:
Renamed.